src/Mapper/ORM/DoctrineORMMapper.php

```diff
@@ -256,8 +257,21 @@ private function loadDiscriminatorColumns(ClassMetadata $metadata): void
             if (isset($this->discriminatorColumns[$metadata->getName()])) {
                 $arrayDiscriminatorColumns = $this->discriminatorColumns[$metadata->getName()];
                 if (isset($metadata->discriminatorColumn)) {
-                    $arrayDiscriminatorColumns = array_merge($metadata->discriminatorColumn, $this->discriminatorColumns[$metadata->getName()]);
+                    $metadataDiscriminatorColumnAsArray = $metadata->discriminatorColumn instanceof DiscriminatorColumnMapping
+                        ? [
```

Comment:
@VincentLanglet you mean moving something like this as `toArray()` into ORM?

Reply:
Why not just cast the object to an array?

Reply:
Didn't think about it. If it works it's perfect indeed

Reply:
It works fine :+1: however static analysis is not happy because for example `length` should not be null